src/julia-parser.scm

```diff
@@ -1141,6 +1141,13 @@
     (begin0 (list 'type (if mut? 'true 'false) sig (parse-block s))
             (expect-end s word))))
 
+;; consume a single line ending from a token stream
+(define (take-lineending s)
+  (let ((nt (peek-token s)))
+    (if (or (newline? nt) (eqv? nt #\;))
+        (take-token s))
+    s))
+
```

Comment:
Actually, it occurs to me that this should be
```scm
(define (take-lineendings s)
  (let ((nt (peek-token s)))
        (if (or (newline? nt) (eqv? nt #\;))
            (begin (take-token s)
                   (take-lineendings s))
            s)))
```
since we should allow any number of line endings or semicolons before the `end`, and there should be a test case for this.